src/grpc/stream_infer_handler.cc

```diff
@@ -163,6 +236,24 @@ ModelStreamInferHandler::Process(InferHandler::State* state, bool rpc_ok)
           requested_model_version);
     }
 
+    if (err == nullptr && header_forward_pattern_ != "") {
+      const auto& metadata = state->context_->ctx_->client_metadata();
+      for (const auto& pair : metadata) {
+        auto& key = pair.first;
+        auto& value = pair.second;
+        std::regex base_regex(header_forward_pattern_);
+        if (std::regex_search(key.data(), base_regex)) {
+          std::string param_key = std::string(key.begin(), key.end());
+          std::string param_value = std::string(value.begin(), value.end());
+          err = TRITONSERVER_InferenceRequestSetStringParameter(
+              irequest, param_key.c_str(), param_value.c_str());
+          if (err != nullptr) {
+            break;
+          }
+        }
+      }
+    }
```

Comment:
Can be a helper function in InferHandler?

Reply:
Fixed.